src/scss/grommet-core/_objects.form-field.scss

```diff
@@ -92,6 +92,12 @@ $colored-select-drop-caret: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAB
       padding: 0 $form-horizontal-padding;
     }
 
+    &.#{$grommet-namespace}search-input__input {
+      // adjust padding for search-input so input doesn't
+      // overlay search icon
+      padding-right: round($inuit-base-spacing-unit * 2);
```

Comment:
The non-FormField context doesn't have the left/right padding overridden by the style above. I've updated the codepen to illustrate the same component outside of a FormField.